debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/YbExporterConsumer.java

```diff
@@ -80,6 +93,97 @@ void connect() throws URISyntaxException {
         flusherThread = new Thread(this::flush);
         flusherThread.setDaemon(true);
         flusherThread.start();
+
+        // Register shutdown hook to release lock on exit
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            releaseLockFile();
+        }));
+    }
+
+    /**
+     * Reads the lock file and checks if the PID is still running.
+     * If the PID is not running, it deletes the lock file.
+     * If the PID is running, it throws an IllegalStateException.
+     */
+    private void readLockFileAndCheckPid() {
```

Comment:
nit: `checkExistingLockFile`
you can move the isLockfileExists block inside this